nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala

```diff
@@ -0,0 +1,132 @@
+       bit of each byte is 1 if the referenced page is in memory, otherwise
+       it is zero.  */
+  def mincore(start: Ptr[Byte], len: CSize, vec: Ptr[Byte]): CInt = extern
+
+  /* Remap pages mapped by the range [ADDR,ADDR+OLD_LEN) to new length
+       NEW_LEN.  If MREMAP_MAYMOVE is set in FLAGS the returned address
+       may differ from ADDR.  If MREMAP_FIXED is set in FLAGS the function
+       takes another parameter which is a fixed address at which the block
+       resides after a successful call.  */
+  def mremap(addr: Ptr[Byte],
+             old_len: CSize,
+             new_len: CSize,
+             flags: CInt): Ptr[Byte] = extern
+
+  /* Remap arbitrary pages of a shared backing store within an existing
+     VMA.  */
+  def remap_file_pages(start: Ptr[Byte],
+                       ize: CSize,
+                       prot: CInt,
+                       pgoff: CSize,
+                       flags: CInt): CInt = extern
+
+  /* Open shared memory segment.  */
+  def shm_open(name: CString, oflag: CInt, mode: mode_t): CInt = extern
+
+  /* Remove shared memory segment.  */
+  def shm_unlink(name: CString): CInt = extern
+
+  // Macros
+  @name("scalanative_prot_exec")
+  def PROT_EXEC: CInt = extern
+  @name("scalanative_prot_read")
+  def PROT_READ: CInt = extern
+  @name("scalanative_prot_write")
+  def PROT_WRITE: CInt = extern
+  @name("scalanative_prot_none")
+  def PROT_NONE: CInt = extern
+  @name("scalanative_map_shared")
+  def MAP_SHARED: CInt = extern
+  @name("scalanative_map_private")
+  def MAP_PRIVATE: CInt = extern
+  @name("scalanative_map_32bit")
+  def MAP_32BIT: CInt = extern
+  @name("scalanative_map_anon")
+  def MAP_ANON: CInt = extern
+  @name("scalanative_map_anonymous")
+  def MAP_ANONYMOUS: CInt = extern
+  @name("scalanative_map_denywrite")
+  def MAP_DENYWRITE: CInt = extern
+  @name("scalanative_map_executable")
+  def MAP_EXECUTABLE: CInt = extern
+  @name("scalanative_map_file")
+  def MAP_FILE: CInt = extern
+  @name("scalanative_map_fixed")
+  def MAP_FIXED: CInt = extern
+  @name("scalanative_map_growsdown")
+  def MAP_GROWSDOWN: CInt = extern
+  @name("scalanative_map_hugetlb")
+  def MAP_HUGETLB: CInt = extern
+  @name("scalanative_map_huge_2mb")
+  def MAP_HUGE_2M: CInt = extern
+  @name("scalanative_map_huge_1gb")
+  def MAP_HUGE_1GB: CInt = extern
+  @name("scalanative_map_huge_shift")
+  def MAP_HUGE_SHIFT: CInt = extern
+  @name("scalanative_map_locked")
+  def MAP_LOCKED: CInt = extern
+  @name("scalanative_map_nonblock")
+  def MAP_NONBLOCK: CInt = extern
+  @name("scalanative_map_noreserve")
+  def MAP_NORESERVE: CInt = extern
+  @name("scalanative_map_populate")
+  def MAP_POPULATE: CInt = extern
+  @name("scalanative_map_stack")
+  def MAP_STACK: CInt = extern
+  @name("scalanative_map_uninitialized")
+  def MAP_UNINITIALIZED: CInt = extern
+  @name("scalanative_config_mmap_allow_uninitialized")
+  def CONFIG_MMAP_ALLOW_UNINITIALIZED: CInt = extern
```

Comment:
Please double check that none of those are linux-specific. 

For example MAP_HUGE_* are not in http://pubs.opengroup.org/onlinepubs/9699919799/